### ISSUE TITLE
Add support for nodejs v20

### ISSUE THREAD

**Issue by mshustov** [ClickHouse/clickhouse-js#149]
Initial Release: 2023-04-18. see https://github.com/nodejs/Release

**Comment by slvrtrn** [comment]
Let's combine it with #148

**Comment by olexiyb** [comment]
In our project we need to use latest node 19(20) as it gives very noticeable performance improvements.
When we started to use node 19 we started to capture Timeout errors on slow queries (we did not set any timeouts).
I was able to track down this to incorrect value in socket.timeout was set to 3000
![image](https://user-images.githubusercontent.com/352780/234514837-b4ad927a-d7d0-4f2a-8893-f16108be9de1.png)
I suspect this is somehow related to https://github.com/nodejs/node/issues/47137

After I have added

```
      request.once('socket', (socket) => {
        socket.setTimeout(this.config.request_timeout)
      })
```
The slow function started to work. This issue was reproducible on Mac M1 and alpine 3.17 (running in docker container)
